vcd/resource_vcd_vapp_network.go

```diff
@@ -271,17 +448,92 @@ func resourceVappNetworkDelete(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error finding vApp: %#v", err)
 	}
 
-	task, err := vapp.RemoveIsolatedNetwork(d.Get("name").(string))
+	_, err = vapp.RemoveNetwork(d.Id())
 	if err != nil {
 		return fmt.Errorf("error removing vApp network: %#v", err)
 	}
 
-	err = task.WaitTaskCompletion()
+	d.SetId("")
+
+	return nil
+}
+
+// resourceVcdVappNetworkImport is responsible for importing the resource.
+// The following steps happen as part of import
+// 1. The user supplies `terraform import _resource_name_ _the_id_string_` command
+// 2. `_the_id_string_` contains a dot formatted path to resource as in the example below
+// 3. The functions splits the dot-formatted path and tries to lookup the object
+// 4. If the lookup succeeds it sets the ID field for `_resource_name_` resource in statefile
+// (the resource must be already defined in .tf config otherwise `terraform import` will complain)
+// 5. `terraform refresh` is being implicitly launched. The Read method looks up all other fields
+// based on the known ID of object.
+//
+// Example resource name (_resource_name_): vcd_vapp_network.network_name
+// Example import path (_the_id_string_): org-name.vdc-name.vapp-name.network-name
+func resourceVcdVappNetworkImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("[vApp network import] resource name must be specified as org-name.vdc-name.vapp-name.network-name")
+	}
+	orgName, vdcName, vappName, networkName := resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3]
+
+	vcdClient := meta.(*VCDClient)
+	_, vdc, err := vcdClient.GetOrgAndVdc(orgName, vdcName)
 	if err != nil {
-		return fmt.Errorf("error waiting for task to complete: %+v", err)
+		return nil, fmt.Errorf("[VM import] unable to find VDC %s: %s ", vdcName, err)
 	}
 
-	d.SetId("")
+	vapp, err := vdc.GetVAppByName(vappName, false)
+	if err != nil {
+		return nil, fmt.Errorf("[VM import] error retrieving vapp %s: %s", vappName, err)
+	}
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return nil, fmt.Errorf("[VM import] error retrieving vApp network configuration %s: %s", networkName, err)
+	}
 
-	return nil
+	vappNetworkToImport := types.VAppNetworkConfiguration{}
+	for _, networkConfig := range vAppNetworkConfig.NetworkConfig {
+		if networkConfig.NetworkName == networkName {
+			vappNetworkToImport = networkConfig
+			break
+		}
+	}
+
+	if vappNetworkToImport == (types.VAppNetworkConfiguration{}) {
```

Comment:
Can we validate that this is a vApp network and not an Org network?

Reply:
not so easy - I will check if it possible

Reply:
added validation